src/utils/classes/models-interfaces.ts

```diff
@@ -0,0 +1,21 @@
+import { PlayNotes } from "../../consts/playNotes";
+import { PlayFillOn, TimeSignature } from "./patternMaker";
+
+export type SubBeatPosition = '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7';
+
+export enum MetronomeSounds {
+	MetronomeSoundOff = '0',
+	MetronomeSoundHi = '1',
+	MetronomeSoundLo = '2',
+	MetronomeSoundFill = '3'
+}
+
+export interface IPatternSettings {
+  playNotes: PlayNotes;
+  playFillOn: PlayFillOn;
```

Comment:
Can we make playFillOn optional? OR make a new interface that inherits from this interface with playFillOn as an extra? not sure of the best option here.

The reason being it isn't used in pattern.ts - it's only used in patternFactory. I don't think we should have unimplemented features in the parent class. The latter of the two options above would be my preference if its possible.